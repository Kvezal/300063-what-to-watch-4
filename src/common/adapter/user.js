const adaptUser = (user) => ({
  id: user.id,
  avatar: `https://4.react.pages.academy${user.avatar_url}`,
  email: user.email,
  name: user.name,
});

export default adaptUser;
