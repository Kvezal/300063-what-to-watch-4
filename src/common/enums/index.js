import FilmOverviewTabsEnum from "./film-overview-tabs";
import GenreEnum from "./genres";


export {
  FilmOverviewTabsEnum,
  GenreEnum,
};
