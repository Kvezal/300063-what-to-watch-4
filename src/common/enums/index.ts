import EFilmTab from "./film-overview-tab";
import EGenre from "./genre";


export {
  EFilmTab,
  EGenre,
};
