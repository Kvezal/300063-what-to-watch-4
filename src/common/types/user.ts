export interface IUser {
  id: number;
  avatar: string;
  email: string;
  name: string;
}

export interface IServerUser {
  id: number;
  email: string;
  name: string;
  avatar_url: string;
}
