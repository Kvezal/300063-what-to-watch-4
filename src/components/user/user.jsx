import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";


const User = (props) => {
  const {avatar, isAuthorized} = props;
  return <div className="user-block">
    {isAuthorized
      ? <div className="user-block__avatar">
        <img src={avatar} alt="User avatar" width="63" height="63"/>
      </div>
      : <Link to="/login" className="user-block__link">Sign in</Link>
    }
  </div>;
};

User.propTypes = {
  avatar: PropTypes.string,
  isAuthorized: PropTypes.bool.isRequired,
};

export default User;
