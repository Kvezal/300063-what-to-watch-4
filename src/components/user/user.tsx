import * as React from "react";
import {Link} from "react-router-dom";

import EAppRoute from "@app/app-route";

import {IUserProps} from "./interface";


const User: React.FC<IUserProps> = (props: IUserProps) => {
  const {avatar, isAuthorized} = props;
  return <div className="user-block">
    {isAuthorized
      ? <Link to={EAppRoute.MY_LIST}>
        <div className="user-block__avatar">
          <img src={avatar} alt="User avatar" width="63" height="63"/>
        </div>
      </Link>
      : <Link to={EAppRoute.LOGIN} className="user-block__link">Sign in</Link>
    }
  </div>;
};

export default User;
