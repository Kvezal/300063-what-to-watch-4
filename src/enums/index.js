import GenreEnum from "./genres";


export {
  GenreEnum,
};
