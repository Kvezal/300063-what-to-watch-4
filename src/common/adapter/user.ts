import {ServerUserInterface, UserInterface} from "@common/types";


const adaptUser = (user: ServerUserInterface): UserInterface => ({
  id: user.id,
  avatar: `https://4.react.pages.academy${user.avatar_url}`,
  email: user.email,
  name: user.name,
});

export default adaptUser;
