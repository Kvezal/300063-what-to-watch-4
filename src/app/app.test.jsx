import React from "react";
import {Provider} from "react-redux";
import renderer from "react-test-renderer";
import {MemoryRouter} from "react-router-dom";
import configureStore from 'redux-mock-store';

import {GenreEnum} from "@common/enums";
import NameSpace from "@store/name-space";
import {AuthorizationStatus} from "@store/user/const";

import App from "./app";


const mockStore = configureStore([]);

const films = [
  {
    id: 1,
    name: `name 1`,
    genre: `Crime`,
    runTime: `3h 49m`,
    releaseDate: 1984,
    description: `description 1`,
    director: `director 1`,
    rating: {
      score: 9.9,
      count: 276395,
      assessment: `Very good`,
    },
    source: {
      video: `video1`,
      previewVideo: `previewVideo1`,
    },
    picture: {
      poster: `poster1`,
      cover: `cover1`,
      backgroundColor: `color1`,
    },
    starring: [
      `Robert De Niro`,
      `James Woods`,
      `Elizabeth McGovern`
    ],
  },
  {
    id: 2,
    name: `name 2`,
    genre: `Crime`,
    runTime: `3h 49m`,
    releaseDate: 1984,
    description: `description 2`,
    director: `director 2`,
    rating: {
      score: 9.9,
      count: 276395,
      assessment: `Very good`,
    },
    source: {
      video: `video2`,
      previewVideo: `previewVideo2`,
    },
    picture: {
      poster: `poster2`,
      cover: `cover2`,
      backgroundColor: `color2`,
    },
    starring: [
      `Robert De Niro`,
      `James Woods`,
      `Elizabeth McGovern`
    ],
  },
  {
    id: 3,
    name: `name 3`,
    genre: `Crime`,
    runTime: `3h 49m`,
    releaseDate: 1984,
    description: `description 3`,
    director: `director 3`,
    rating: {
      score: 9.9,
      count: 276395,
      assessment: `Very good`,
    },
    source: {
      video: `video3`,
      previewVideo: `previewVideo3`,
    },
    picture: {
      poster: `poster3`,
      cover: `cover3`,
      backgroundColor: `color3`,
    },
    starring: [
      `Robert De Niro`,
      `James Woods`,
      `Elizabeth McGovern`
    ],
  },
  {
    id: 4,
    name: `name 4`,
    genre: `Crime`,
    runTime: `3h 49m`,
    releaseDate: 1984,
    description: `description 4`,
    director: `director 4`,
    rating: {
      score: 9.9,
      count: 276395,
      assessment: `Very good`,
    },
    source: {
      video: `video4`,
      previewVideo: `previewVideo4`,
    },
    picture: {
      poster: `poster4`,
      cover: `cover4`,
      color: `color4`,
    },
    starring: [
      `Robert De Niro`,
      `James Woods`,
      `Elizabeth McGovern`
    ],
  },
];

const reviews = [
  {
    text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    ratingScore: 8.9,
    author: `Kate Muir`,
    date: `December 24, 2016`,
  },
  {
    text: `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
    ratingScore: 8.0,
    author: `Bill Goodykoontz`,
    date: `November 18, 2015`,
  },
  {
    text: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
    ratingScore: 8.0,
    author: `Amanda Greever`,
    date: `November 18, 2015`,
  }
];

describe(`App`, () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      [NameSpace.DATA]: {
        genre: GenreEnum.All,
        films,
        promoFilm: films[0],
        currentFilmId: 1,
        reviews,
        notifications: [],
      },
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.AUTH,
      }
    });
  });

  it(`should match to snapshot`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <MemoryRouter>
              <App/>
            </MemoryRouter>
          </Provider>,
          {createNodeMock: () => ({})}
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
