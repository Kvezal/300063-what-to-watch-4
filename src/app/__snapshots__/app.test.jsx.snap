// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match to snapshot 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="name 1"
        src="cover1"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="name 1"
            height="327"
            src="poster1"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            name 1
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Crime
            </span>
            <span
              className="movie-card__year"
            >
              1984
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
          >
            Comedies
          </a>
        </li>
        <li
          className="catalog__genres-item"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
          >
            Crime
          </a>
        </li>
        <li
          className="catalog__genres-item"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
          >
            Documentary
          </a>
        </li>
        <li
          className="catalog__genres-item"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
          >
            Dramas
          </a>
        </li>
        <li
          className="catalog__genres-item"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
          >
            Horror
          </a>
        </li>
        <li
          className="catalog__genres-item"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
          >
            Kids & Family
          </a>
        </li>
        <li
          className="catalog__genres-item"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
          >
            Romance
          </a>
        </li>
        <li
          className="catalog__genres-item"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
          >
            Sci-Fi
          </a>
        </li>
        <li
          className="catalog__genres-item"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
          >
            Thrillers
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      />
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
  <ul
    className="notification-list"
  />,
]
`;
